Upstream status: Fix included starting 9.0-RELEASE

Commit message:
Author: eri <eri@FreeBSD.org>
Date:   Fri Mar 4 20:37:38 2011 +0000

    Fix a panic that can happen when trying to destroy a lagg(4) with scheduler set to none.
    
    Approved by:        thompsa(mentor)
    MFC after:  1 week

Index: sys/net/if_lagg.c
===================================================================
RCS file: /home/ncvs/src/sys/net/if_lagg.c,v
retrieving revision 1.37.2.3.2.1
diff -u -r1.37.2.3.2.1 if_lagg.c
--- sys/net/if_lagg.c	14 Jun 2010 02:09:06 -0000	1.37.2.3.2.1
+++ sys/net/if_lagg.c	3 Mar 2011 16:20:36 -0000
@@ -274,7 +274,8 @@
 	while ((lp = SLIST_FIRST(&sc->sc_ports)) != NULL)
 		lagg_port_destroy(lp, 1);
 	/* Unhook the aggregation protocol */
-	(*sc->sc_detach)(sc);
+	if (sc->sc_detach != NULL)
+		(*sc->sc_detach)(sc);
 
 	LAGG_WUNLOCK(sc);
 
